src/demandlib/vdi/regions.py

```diff
@@ -0,0 +1,598 @@
+using the annuity method defined in the German VDI 4655.
+
+    **VDI 4655**
+
+    **Reference load profiles of single-family and
+    multi-family houses for the use of CHP systems**
+
+    *May 2008 (ICS 91.140.01)*
+
+    *Verein Deutscher Ingenieure e.V.*
+
+    *VDI Standards Department*
+
+    *VDI-Platz 1, 40468 Duesseldorf, Germany*
+
+Reproduced with the kind permission of the Verein Deutscher Ingenieure e.V.
+
+Notes
+-----
+This script creates full year energy demand time series of domestic buildings
+for use in simulations. This is achieved by an implementation of the VDI 4655,
+which gives sample energy demands for a number of typical days ('Typtage').
+The energy demand contains heating, hot water and electricity.
+
+For a given year, the typical days can be matched to the actual calendar days,
+based on the following conditions:
+    - Season: summer, winter or transition
+    - Day: weekday or sunday (Or holiday, which counts as sunday)
+    - Cloud coverage: cloudy or not cloudy
+    - House type: single-family houses or multi-family houses (EFH or MFH)
+
+The holidays are loaded from an Excel file, the weather conditions are loaded
+from weather data (e.g. DWD TRY).
+Most of the settings for this script are controlled with a configuration file
+called 'config_file', the location of which is defined down below.
+
+SPDX-FileCopyrightText: Joris Zimmermann
+SPDX-FileCopyrightText: Uwe Krien
+
+SPDX-License-Identifier: MIT
+"""
+
+import calendar
+import datetime
+import os
+import warnings
+from collections import namedtuple
+
+import pandas as pd
+
+from demandlib.tools import add_weekdays2df
+from demandlib.vdi import dwd_try
+
+
+class Region:
+    def __init__(
+        self,
+        year,
+        try_region,
+        seasons=None,
+        holidays=None,
+        houses=None,
+        resample_rule=None,
+    ):
+        """
+
+        Parameters
+        ----------
+        year : int
+            Year of the profile.
+        try_region : int
+            The number of the TRY (test reference year of the DWD) region.
+        seasons : dict (optional)
+            The times of the seasons if fixed seasons are used. In the VDI
+            norm seasons are defined by the daily average temperature: Winter
+            below 5 degree Celsius, Spring between 5 and 15 degree Celsius and
+            summer above 15 degree Celsius.
+        holidays : dict or list
+            In case of a dictionary the keys are datetime objects and the
+            values are strings with the name of the holiday. Otherwise a list
+            of datetime objects should be passed.
+        houses : list
+            A list of dictionaries in which each house is defined the
+            dictionary need to have the following keys.
+        resample_rule : str
+            Time interval to resample the profile e.g. 1H (1 hour) or 15min.
+            The value will be passed to the pandas resample method.
```

Comment:
```suggestion
            The value will be passed to the pandas resample method.
        file_weather : str (optional)
            Path to a 'test reference year' (TRY) weather file by German DWD
            (Deutscher Wetterdienst). If None, the file fitting the given
            try_region will be loaded.
```

Reply:
As long as you adhere to the TRY format, the actual file does not need to be provided by DWD.